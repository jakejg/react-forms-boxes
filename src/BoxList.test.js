import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';


it('renders without crashing', ()=> {
    render(<BoxList />)
});

it('matches snapshot', () => {
  const { asFragment } = render(<BoxList />)
  expect(asFragment()).toMatchSnapshot();
})

it('keeps track of form data', () => {
  const { getByLabelText} = render(<BoxList />);
  const widthInput = (getByLabelText("width"));
  fireEvent.change(widthInput, {target: {value: 2}})

  expect(widthInput.value).toEqual("2")
})


it('input fields clear after submit', () => {
  const { getByLabelText, getByText} = render(<BoxList />);
  const widthInput = getByLabelText("width");
  const btn = getByText("Create a Box");
  fireEvent.change(widthInput, {target: {value: 2}});
  fireEvent.click(btn)

  expect(widthInput.value).toEqual("")

})

it('it should adds a box to the DOM', () => {
  const { getByLabelText, getByText} = render(<BoxList />);
  const widthInput = getByLabelText("width");
  const heightInput = getByLabelText("height");
  const colorInput = getByLabelText("backgroundColor");
  const btn = getByText("Create a Box");
  fireEvent.change(widthInput, {target: {value: 2}});
  fireEvent.change(heightInput, {target: {value: 2}});
  fireEvent.change(colorInput, {target: {value: "blue"}});
  fireEvent.click(btn)  

  expect(getByText("X").previousSibling).toHaveStyle(`
  width: 2px;
  height: 2px;
  background-color: blue;
  `)

})

it('should delete a box from the DOM', () => {
  const { getByLabelText, getByText} = render(<BoxList />);
  const widthInput = getByLabelText("width");
  const heightInput = getByLabelText("height");
  const colorInput = getByLabelText("backgroundColor");
  const createBoxBtn = getByText("Create a Box");
  fireEvent.change(widthInput, {target: {value: 2}});
  fireEvent.change(heightInput, {target: {value: 2}});
  fireEvent.change(colorInput, {target: {value: "blue"}});
  fireEvent.click(createBoxBtn);

  const box = getByText("X").previousSibling
  const deleteBtn = getByText("X");
  fireEvent.click(deleteBtn)

  expect(box).not.toBeInTheDocument;

});
