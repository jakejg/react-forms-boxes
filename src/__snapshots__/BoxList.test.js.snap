// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="width"
        >
          width
        </label>
        <input
          id="width"
          name="width"
          type="text"
          value=""
        />
        <br />
      </div>
      <div>
        <label
          for="height"
        >
          height
        </label>
        <input
          id="height"
          name="height"
          type="text"
          value=""
        />
        <br />
      </div>
      <div>
        <label
          for="backgroundColor"
        >
          backgroundColor
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          type="text"
          value=""
        />
        <br />
      </div>
      <button>
        Create a Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
